src/utils/github-sync.ts

```diff
@@ -1,122 +1,142 @@
+import git from "isomorphic-git"
+import http from "isomorphic-git/http/web"
+import { fs } from "./fs"
+import { logFileSystemState } from "./log-file-system"
 
 // Store SHA to avoid re-fetching notes if the SHA hasn't changed
 export const shaAtom = atomWithStorage<string | null>("sha", null)
 
 const githubUserCallback = (get: Getter) => get(githubUserAtom)
-const githubRepoCallback = (get: Getter) => get(githubRepoAtom)
-
 const notesCallback = (get: Getter) => get(notesAtom)
 
 export function useRenameTag() {
   const getGitHubUser = useAtomCallback(githubUserCallback)
-  const getGitHubRepo = useAtomCallback(githubRepoCallback)
   const getNotes = useAtomCallback(notesCallback)
   const setRawNotes = useSetAtom(rawNotesAtom)
 
   return React.useCallback(
     async (oldName: string, newName: string) => {
       const notes = getNotes()
 
-      const updatedRawNotes = [...notes.values()]
-        // Notes that contain the old tag
-        .filter((note) => note.tags.includes(oldName))
-        .reduce<Record<string, string>>((updatedRawNotes, note) => {
-          // Find and replace the old tag with the new tag
-          updatedRawNotes[note.id] = note.content.replace(`#${oldName}`, `#${newName}`)
-          return updatedRawNotes
-        }, {})
+      const updatedNotesContent: Record<string, string> = {}
+      for (const note of notes.values()) {
+        if (note.tags.includes(oldName)) {
+          const newContent = note.content.replace(`#${oldName}`, `#${newName}`)
+          updatedNotesContent[note.id] = newContent
 
-      // Update state
-      setRawNotes((rawNotes) => ({ ...rawNotes, ...updatedRawNotes }))
+          // Write updated content to fs
+          await fs.promises.writeFile(`${REPO_DIR}/${note.id}.md`, newContent, "utf8")
+          await git.add({ fs, dir: REPO_DIR, filepath: `${note.id}.md` })
+        }
+      }
 
       // Push to GitHub
       try {
         const githubUser = getGitHubUser()
-        const githubRepo = getGitHubRepo()
-        if (!githubUser || !githubRepo) return
+        if (!githubUser) return
+
+        await git.commit({
+          fs,
+          dir: REPO_DIR,
+          message: `Rename tag #${oldName} to #${newName}`,
+          author: {
+            name: githubUser.name,
+            email: githubUser.email,
+          },
+        })
 
-        const files = mapObject(updatedRawNotes, (content, id) => {
-          return [`${id}.md`, content]
+        await git.push({
+          fs,
+          http,
+          dir: REPO_DIR,
+          onAuth: () => ({ username: githubUser.login, password: githubUser.token }),
         })
 
-        await writeFiles({
-          githubToken: githubUser.token,
-          githubRepo,
-          files,
-          commitMessage: `Rename tag #${oldName} to #${newName}`,
+        setRawNotes((rawNotes) => {
+          return { ...rawNotes, ...updatedNotesContent }
         })
+
+        await logFileSystemState(fs)
```

Comment:
Should we only log the fs state in DEV mode?

Reply:
totally, and this isn't a good place to log it anyway.